### PR TITLE
Added New Provider for Afghanistan [fa_AF]

### PR DESCRIPTION
**Pull request by absherzad**
I added Person, Internet, Company, Text, Address and PhoneNumber providers for Afghanistan [fa_AF]. These are the contribution:
- 1000 real male FirstNames commonly are used in Afghanistan in Persian/Dari format,
- 1000 real female FirstNames commonly are used in Afghanistan in Persian/Dari format,
- 1000 real LastNames commonly are used in Afghanistan in Persian/Dar formati,
- 1000 real male and female FirstName commonly are used in Afghanistan in ASCII format,
- 1000 real LastName commonly are used in Afghanistan in ASCII format,
- 420 real Educational institutional names commonly are used in Afghanistan in Persian/Dari format,
- etc.

### CODE REVIEW

**Comment by absherzad** [comment]
Additionally, the file ReadMe file updated for [fa_AF] examples ...

File: src/Faker/Provider/fa_AF/Company.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+namespace Faker\Provider\fa_AF;
+
+class Company extends \Faker\Provider\Company
+{
+    protected static $formats = array(
+        '{{companyName}}',
+    );
+
+    /**
+     * @source Afghanistan Kankor (University Entrance Exam) 2003-2017 data.
```
Comment (fzaninotto) [inline src/Faker/Provider/fa_AF/Company.php]: not enough, you must add a link to a source I can verify. same for other formatters with the same source

File: src/Faker/Provider/fa_AF/Internet.php [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+    protected static $tld = array(
+        'com', 'gov', 'org', 'af', 'com.af', 'net', 'info', 'edu.af', 'gov.af',
+    );
+
+    /**
+     * @source Afghanistan Kankor (University Entrance Exam) 2003-2017 data.
+     * The data collected, preprocessed and prepared by Abdul Rahman Sherzad (absherzad@gmail.com).
+     * @license: Creative Commons Attribution-ShareAlike License
+    */
+    protected static $lastNameAscii = array("Ahmadi", "Mohammadi", "Noori", "Rahimi", "Amiri", "Haidari", "Azizi", "Siddiqi", "Hosseini", "Nazari", "Karimi", "Rahmani", "Hashemi",
```
Comment (fzaninotto) [inline src/Faker/Provider/fa_AF/Internet.php]: should be sorted alphabetically and split by line for each new capital letter (see formatters in other languages for a guide)

File: src/Faker/Provider/fa_AF/Text.php [hunk 3]
```diff
@@ -0,0 +1,546 @@
...
+    /**
+     * generates text string in arabic
+     *
+     * @example 'از تاریخ‌الشعرا را بکوبند روی نبش دیوار کوچه‌شان. تابلوی مدرسه.'
+     * @param  integer                   $maxNbChars
+     * @param  integer                   $indexSize
+     * @return string
+     * @throws \InvalidArgumentException
+     */
+    public function realText($maxNbChars = 200, $indexSize = 2)
```
Comment (fzaninotto) [inline src/Faker/Provider/fa_AF/Text.php]: why do you need to override the original method?

File: src/Faker/Provider/fa_AF/Company.php (hunk 1 above)
Comment (absherzad) [reply src/Faker/Provider/fa_AF/Company.php]: I collected the Kankor data for my PhD research purposes, and still I did not publish these data anywhere. But I add a link to one of my paper that confirms it.

File: src/Faker/Provider/fa_AF/Text.php (hunk 3 above)
Comment (absherzad) [reply src/Faker/Provider/fa_AF/Text.php]: The added regular expression almost includes all the required and common characters in Persian/Dari text including different varieties of quotes and apostrophes, digits in Arabic and Persian formats,  zero width space characters, and others.

File: src/Faker/Provider/fa_AF/Company.php (hunk 1 above)
Comment (fzaninotto) [reply src/Faker/Provider/fa_AF/Company.php]: There is no way to check that the data actually comes from there, as it's not published in your paper...

Comment (absherzad) [reply src/Faker/Provider/fa_AF/Company.php]: I mean the content of the paper confirms and verifies the Kankor data collection ...

Comment (fzaninotto) [reply src/Faker/Provider/fa_AF/Company.php]: I need a link to the actual collection.

Comment (absherzad) [reply src/Faker/Provider/fa_AF/Company.php]: Link added ...

**Comment by fzaninotto** [comment]
There are codestyle issues (see GitHub checks details)

Alss, I'd prefer datasets of about 300 items instead of 1000

File: readme.md [hunk 4]
```diff
@@ -1003,6 +1003,71 @@ echo $faker->vat;           // "A35864370"
 echo $faker->dni; // '83367512'
 ```
 
+### `Faker\Provider\fa_AF\Person`
+
+```php
+<?php
```
Comment (fzaninotto) [inline readme.md]: Please mention only the formatters specific to your locale, so e.g. remove `lastName` as it's already documented in the main `Person` section.

Comment (absherzad) [reply readme.md]: I fixed the locale formatters in the readme file. Please let me know if any further modifications are needed.

Comment (fzaninotto) [reply readme.md]: Most of formatters you mentioned in the locale-specific section exist in every locale (see https://github.com/fzaninotto/Faker/blob/master/src/Faker/Generator.php for the list). You should only document those that are specific to your locale : `firstNameAscii`, `lastNameAscii`, and `cityName`.

Also, there are some non-ascii characters in the output of the `firstNameAscii` documentation:

image

**Comment by fzaninotto** [comment]
see Travis CI build report: you have formatting errors in your code

**Comment by absherzad** [comment]
Can you please give me some hints how can I fix the formatting errors. I did not find useful info following "The Travis CI build failed" details section.

**Comment by ppelgrims** [comment]
@absherzad Did you run `make sniff test`? It runs phpcs (PHP_CodeSniffer), which will outline the problems with formatting, that package also includes phpcbf which can fix it automagically.

Across most files: `ERROR | Whitespace found at end of line`
In `fa_AF/Company.php`: `36 | ERROR | The closing brace for the class must go on the next line after the body`

File: src/Faker/Provider/fa_AF/Internet.php [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+    public static function firstNameAscii()
+    {
+        return static::randomElement(static::$firstNameAscii);
+    }
+
+    /**
+     * Only getting first character of fristName
+     */
+    public static function firstNameAsciiFirstChar()
```
Comment (fzaninotto) [inline src/Faker/Provider/fa_AF/Internet.php]: please remove this one, it has no aded value, plus it fails on Travis

**Comment by fzaninotto** [comment]
tests still fail

**Comment by absherzad** [comment]
Could you please be more specific what changes are required to be made?

**Comment by pimjansen** [comment]
Bump!
@absherzad could you please review the build and given comments? 

Thanks a lot

**Comment by pimjansen** [comment]
I am going to close this PR as stale. If this still needs to be merged, please feel free to reopen it.
Thanks for contributing

**Pull request closed without merging**